packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -12,6 +15,105 @@ import 'gesture_detector.dart';
 import 'navigator.dart';
 import 'transitions.dart';
 
+// Recognizes tap down by any pointer button unconditionally. When it receives a
```

Comment:
Maybe move these helper classes to the end of the file since a users coming to this file is probably more interested in the ModalBarrier?